src/components/Scenarios/ScenariosFormField.jsx

```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import AutoComplete from 'material-ui/AutoComplete';
+import getFormFieldData from './FormFieldData';
+import { listStyle } from './Styles';
+import strings from '../../lang';
+
+const hintTexts = {
+  itemTimings: {
+    hero_id: strings.filter_hero_id,
+    item: strings.scenarios_item,
+    time: strings.scenarios_time,
+  },
+  laneRoles: {
+    hero_id: strings.filter_hero_id,
+    lane_role: strings.heading_lane_role,
+    time: strings.scenarios_game_duration,
+  },
+  misc: {
+    scenario: strings.scenarios_scenario,
+  },
+};
+
+const customStyles = {
+  scenario: { width: '450px' },
+};
+
+class ScenarioFormField extends React.Component {
+  constructor(props) {
+    super(props);
+    const {
+      field, formFieldState, metadata, selectedTab,
+    } = this.props;
+    const {
+      heroList, itemList, laneRoleList, miscList, gameDurationList, timingList,
+    } = getFormFieldData(metadata);
+    this.dataSources = {
+      itemTimings: {
+        hero_id: heroList,
+        item: itemList,
+        time: timingList,
+      },
+      laneRoles: {
+        hero_id: heroList,
+        lane_role: laneRoleList,
+        time: gameDurationList,
+      },
+      misc: {
+        scenario: miscList,
+      },
+    };
+
+
+    let searchText = this.dataSources[selectedTab][field].find(el => el.value === formFieldState);
+    searchText = searchText ? searchText.text : '';
+
+    this.state = {
+      searchText,
+    };
+
+    this.resetField = this.resetField.bind(this);
+    this.handleUpdateInput = this.handleUpdateInput.bind(this);
+    this.handleRequest = this.handleRequest.bind(this);
+  }
+
+  componentDidMount() {
+    requestAnimationFrame(() => {
```

Comment:
does this fire many times per second?  might have a perf impact

Reply:
it renders once , skips a frame and then renders each AutoComplete component once again. so they are rendered only one additional time.
I believe the performance loss is negligible.

the frame skip is necessary because browsers merge changes in a single frame together for optimization reasons.